crates/polars-python/src/file.rs

```diff
@@ -184,14 +202,25 @@ impl Write for PyFileLikeObject {
                     None,
                 )
             } else {
+                // tracks the number of bytes written
                 self.inner
                     .call_method(py, "write", (PyBytes::new(py, buf),), None)
             }
             .map_err(pyerr_to_io_err)?;
 
-            let n = number_bytes_written.extract(py).map_err(pyerr_to_io_err)?;
-
-            Ok(n)
+            let n = char_or_bytes_written.extract(py).map_err(pyerr_to_io_err)?;
+            let n_bytes = if expects_str && !expects_str_and_is_ascii {
+                // substitute char count with buf len
+                let expected_chars = std::str::from_utf8(buf)
+                    .map_err(io::Error::other)?
+                    .chars()
+                    .count();
+                assert!(n == expected_chars);
```

Comment:
I think this is not what you want. You want to do is find the offset of the nth char. This would go wrong if you have a rate limited network interface for instance. You can use `.char_indices().nth(n).map(|(i, _)| i).expect("...")` for this.

```